### PR TITLE
embed build number in PR version

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
   <Target Name="AppVeyorPullRequestsTarget" AfterTargets="MinVer" Condition="'$(APPVEYOR_PULL_REQUEST_NUMBER)' != ''">
     <PropertyGroup>
-      <PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch)-PR.$(APPVEYOR_PULL_REQUEST_NUMBER).$(MinVerPreRelease)</PackageVersion>
+      <PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch)-PR.$(APPVEYOR_PULL_REQUEST_NUMBER)-Build.$(APPVEYOR_BUILD_NUMBER)-$(MinVerPreRelease)</PackageVersion>
```
Comment (adamralph) [inline src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj]: Bear in mind that the dash means that, according to SemVer, the pull request number plus the dash plus "Build" will form one of the single dot-separated identifiers.

If you want the build number to be called out explicitly, I would recommed this format (with consistent case thrown in by my OCD 😉):

```suggestion
      <PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch)-pr.$(APPVEYOR_PULL_REQUEST_NUMBER).build.$(APPVEYOR_BUILD_NUMBER)-$(MinVerPreRelease)</PackageVersion>
```

Comment (adamralph) [reply src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj]: Or alternatively, you could have the PR and build labels and their numbers as individual identifiers:

```suggestion
      <PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch)-pr-$(APPVEYOR_PULL_REQUEST_NUMBER).build-$(APPVEYOR_BUILD_NUMBER).$(MinVerPreRelease)</PackageVersion>
```

Comment (mauroservienti) [reply src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj]: Thanks. I like the first one more. I’ll take that.

**Comment by mauroservienti** [comment]
@adamralph the annoying thing is that the APPVEYOR_BUILD_NUMBER is not increased at every build for the same PR. Anyway I understood the intent, I’ll have a look at what AppVeyor offers as variables and play with them a bit to find a good combination.

**Comment by adamralph** [comment]
@mauroservienti maybe what you need is `APPVEYOR_BUILD_ID`. Perhaps:

```xml
<PackageVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch)-pr-$(APPVEYOR_PULL_REQUEST_NUMBER).build-$(APPVEYOR_BUILD_NUMBER).$(APPVEYOR_BUILD_ID).$(MinVerPreRelease)</PackageVersion>
```

File: src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj (hunk 1 above)
Comment (mauroservienti) [reply src/ServiceComposer.AspNetCore/ServiceComposer.AspNetCore.csproj]: Interestingly I cannot commit suggestions. I’ll close this one and commit directly to the original one your suggestion. Thanks.

**Comment by mauroservienti** [comment]
I guess that build number + job number should do the trick. The IDs are all strings randomly generated.

**Comment by adamralph** [comment]
@mauroservienti do you mean `APPVEYOR_JOB_NUMBER`? You only have one job in your build, so I think the job number will always be 1, no?

I guessed that `APPVEYOR_BUILD_ID` would be the 24153311 in https://ci.appveyor.com/project/ServiceComposer/servicecomposer-aspnetcore/builds/24153311 but perhaps I'm wrong. I think that's the number you need though.

**Comment by mauroservienti** [comment]
you're right,  inverted stuff in my mind. Hopefully `APPVEYOR_BUILD_ID` is always increasing and never resetting. In the meantime I cherry picked some of the changes into #18, and verified that it works as expected. Thanks for your time @adamralph, much appreciated.

**Comment by adamralph** [comment]
`APPVEYOR_BUILD_ID` can't ever reset, since it's used in the URL to identify a build.

I think it's reasonable to assume that it's every increasing. I suspect it's a _globally_ unique build ID so incrementing it seems like the easiest approach to keeping it unique.

**Comment by mauroservienti** [comment]
very good point.

**Pull request closed without merging**